src/main/java/com/github/jmchilton/blend4j/galaxy/ToolDataClient.java

```diff
@@ -0,0 +1,25 @@
+package com.github.jmchilton.blend4j.galaxy;
+
+import com.github.jmchilton.blend4j.galaxy.beans.TabularToolDataTable;
+import com.sun.jersey.api.client.ClientResponse;
+
+import java.util.List;
+
+public interface ToolDataClient {
```

Comment:
Could you add the API functions `delete_data_table` and `reload_data_table` in here as well (https://bioblend.readthedocs.io/en/latest/api_docs/galaxy/all.html#module-bioblend.galaxy.tool_data).  The function `reload_data_table` may be needed if databases get updated in Galaxy without restarting Galaxy.  The function `delete_data_table` could be needed if we ever wanted to automate cleanup of databases.

Reply:
How closely should we match the python API? 

```java
public void deleteDataTable(String dataTableId, String values);
```

...where values is "a “|” separated list of column contents, there must be a value for all the columns of the data table"

...or

```java
public void deleteDataTable(String dataTableId, List<String> values);
```

(or both?)

Reply:
I think using a `List` of values is better.  No need to do both.

Reply:
I'm working on this. Having some trouble figuring out how to send the 'values' information to the Galaxy API. I've sent a question out on the galaxy-dev listserv.

Reply:
Thanks for looking into this.  I tested it out myself with bioblend and I also can't quite figure out how to send the 'values'.

Reply:
The test case that I discussed in my galaxy-dev email is to delete the `hg38` entry from the `igv_broad_genomes` tool data table.

The closest I've got so far is this:

```python
from bioblend.galaxy import GalaxyInstance
from bioblend.galaxy.tool_data import ToolDataClient
gi = GalaxyInstance('localhost:48888', key='admin') # (substitute your galaxy url & api key)
tdc = bioblend.galaxy.tool_data.ToolDataClient(gi)
tdc.delete_data_table("igv_broad_genomes", "Human hg38\thttp://s3.amazonaws.com/igv.broadinstitute.org/genomes/hg38.genome\thg38")
```

...that returns the original tool data table, but doesn't seem to remove the hg38 entry. Note that I used `\t` separators, not `|` separators as the api documentation suggests. If I use `|` separators then I get an error code stating:

```
"Invalid data table item ( Human hg38|http://s3.amazonaws.com/igv.broadinstitute.org/genomes/hg38.genome|hg38 ) specified.
Wrong number of columns (1 given, 3 required)."
```

Reply:
Thanks for taking a look at this.  I wonder if running [reload_data_table](https://bioblend.readthedocs.io/en/latest/api_docs/galaxy/all.html#bioblend.galaxy.tool_data.ToolDataClient.reload_data_table) would then show it as deleted.  I know if you add a new table you have to run `reload_data_table` before it will show up as a new entry in the user interface or in the API.

Reply:
I've tried reloading the data table from both bioblend and the web interface, and neither show that the hg38 entry has been deleted.

Reply:
Ah... I think I see why the `hg38` entry isn't removed from the `igv_broad_genomes` table. The `delete` call automatically calls `reload_tool_data_tables` and for the `igv_broad_genomes` the reload involves pulling the table from [this URL](http://igv.broadinstitute.org/genomes/genomes.txt)

https://github.com/galaxyproject/galaxy/blob/222ece4eabb6d080c6b8a0e327f8cf617b8a194f/lib/galaxy/webapps/galaxy/api/tool_data.py#L48

```
...
172.17.0.1 - - [11/Nov/2017:22:44:13 +0000] "DELETE /api/tool_data/igv_broad_genomes?key=admin HTTP/1.1" 200 - "-" "PostmanRuntime/6.4.1"
[pid: 274|app: 0|req: 62/138] 172.17.0.1 () {46 vars in 777 bytes} [Sat Nov 11 22:44:13 2017] DELETE /api/tool_data/igv_broad_genomes?key=admin => generated 17488 bytes in 534 msecs (HTTP/1.1 200) 3 headers in 124 bytes (1 switches on core 2)
galaxy.queue_worker INFO 2017-11-11 22:44:14,992 Instance 'main.2' received 'reload_tool_data_tables' task, executing now.
galaxy.queue_worker DEBUG 2017-11-11 22:44:14,993 Executing tool data table reload for all tables
requests.packages.urllib3.connectionpool INFO 2017-11-11 22:44:15,000 Starting new HTTP connection (1): igv.broadinstitute.org
requests.packages.urllib3.connectionpool DEBUG 2017-11-11 22:44:15,218 "GET /genomes/genomes.txt HTTP/1.1" 200 15681
galaxy.tools.data DEBUG 2017-11-11 22:44:15,314 Loading Data Table URL "http://igv.broadinstitute.org/genomes/genomes.txt" as filename "/export/galaxy-central/database/files/TTDT_URL_igv_broad_genomes-GIDO9d".
galaxy.tools.data WARNING 2017-11-11 22:44:15,410 Line 1 in tool data table 'igv_broad_genomes' is invalid (HINT: '<TAB>' characters must be used to separate fields):
<Server-Side Genome List>
galaxy.tools.data DEBUG 2017-11-11 22:44:15,411 Loaded 156 lines from '/export/galaxy-central/database/files/TTDT_URL_igv_broad_genomes-GIDO9d' for 'igv_broad_genomes'
galaxy.tools.data DEBUG 2017-11-11 22:44:15,414 Reloaded tool data table 'igv_broad_genomes' from files.
galaxy.queue_worker DEBUG 2017-11-11 22:44:15,414 Finished data table reload for [u'igv_broad_genomes']
...
```

...so I might be able to run a successful delete API call on a mentalist tool data table, but I've been having trouble recently installing packages via conda. I get this error:

```
CondaError: OSError(40, 'Too many levels of symbolic links')
```

...not sure what's going on there.

Reply:
Relevant StackOverflow:

https://stackoverflow.com/questions/25229880/how-to-send-enclose-data-in-delete-request-in-jersey-client

Reply:
I'm fairly sure I have this working now in these [two](https://github.com/apetkau/blend4j/pull/9/commits/c339208557c6175ae9eaf7806d0e88a6229707b7) [commits](https://github.com/apetkau/blend4j/pull/9/commits/3ab5cc543992d253ff663ae236c666b7abb13b8d).

Reply:
Thanks for taking the time to work on this.